From bbce6d63c18b85710b1e567a3035914c5ac13ccf Mon Sep 17 00:00:00 2001
From: Marcelo Tosatti <mtosatti@redhat.com>
Date: Fri, 29 Jul 2011 19:40:51 -0300
Subject: [PATCH 81/91] seabios: increase smp_mtrr array size

Windows Server 2008 checked build complains about lack of consistency
between MTRR MSRs in SMP guests.

Problem is the smp_mtrr array is not large enough to hold all MSRs (31
entries with current qemu/kvm implementations).

Increase it to 32.

Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>
---
 src/smp.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/smp.c b/src/smp.c
index 40f5451..2317d36 100644
--- a/src/smp.c
+++ b/src/smp.c
@@ -17,7 +17,7 @@
 
 #define APIC_ENABLED 0x0100
 
-struct { u32 ecx, eax, edx; } smp_mtrr[16] VAR16VISIBLE;
+struct { u32 ecx, eax, edx; } smp_mtrr[32] VAR16VISIBLE;
 u32 smp_mtrr_count VAR16VISIBLE;
 
 void
-- 
1.7.5.4

